sw/airborne/modules/uav_recovery/uav_recovery.c

```diff
@@ -0,0 +1,272 @@
+/*
+ * $Id: parachute.c 21 November 2012 by Chris Efstathiou
+ *
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+#include "autopilot.h"
+#include "state.h"
+#include "firmwares/fixedwing/nav.h"
+#include "subsystems/navigation/common_nav.h"
+#include "generated/flight_plan.h"
+#include "generated/airframe.h"
+#include "inter_mcu.h"
+#include "subsystems/datalink/datalink.h"
+#include "modules/multi/traffic_info.h"
+#include "uav_recovery.h"
+
+#if defined(PARACHUTE_SERVO_CHANNEL) && PARACHUTE_SERVO_CHANNEL != -1
+#define PARACHUTE_OUTPUT_COMMAND    CONCAT(COMMAND_, PARACHUTE_SERVO_CHANNEL)
+#else
+#warning YOU HAVE NOT DEFINED A PARACHUTE RELEASE AUTOPILOT SERVO
+#endif
+
+/** TERMINAL VELOCITY OF THE PARACHUTE PLUS PAYLOAD */
+#ifndef PARACHUTE_TRIGGER_DELAY
+#define PARACHUTE_TRIGGER_DELAY 2.
+#endif
+
+#ifndef PARACHUTE_DESCENT_RATE
+#define PARACHUTE_DESCENT_RATE    3.0
+#endif
+#ifndef PARACHUTE_WIND_CORRECTION
+#define PARACHUTE_WIND_CORRECTION 1.0
+#endif
+#ifndef PARACHUTE_LINE_LENGTH
+#define PARACHUTE_LINE_LENGTH   3.0
+#endif
+
+float  parachute_start_qdr;
+float  parachute_z;
+float  airborne_wind_dir = 0;
+float  airborne_wind_speed = 0;
+float  calculated_wind_dir = 0;
+bool   wind_measurements_valid = TRUE;
```

Comment:
It is now recommended to use `true` or `false` instead of uppercase version in the code. It is part of `stdbool.h` included by `std.h`.

Reply:
FIXED